src/test/macromanager_test.cpp

```diff
@@ -0,0 +1,66 @@
+#include "recording/macromanager.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <QDebug>
+
+#include "mixxxtest.h"
+#include "signalpathtest.h"
+
+TEST(MacrosTest, CreateMacro) {
+    auto macro = new Macro();
+    ASSERT_EQ(macro->m_length, 0);
+}
+
+TEST(MacroManagerTest, ClaimRecording) {
+    auto mgr = MacroManager();
+    EXPECT_EQ(mgr.isRecordingActive(), false);
+    mgr.claimRecording();
+    EXPECT_EQ(mgr.isRecordingActive(), false);
+    mgr.m_macroRecordingState.store(MacroState::Armed);
+    mgr.claimRecording();
+    EXPECT_EQ(mgr.isRecordingActive(), true);
+}
+
+TEST(MacroManagerTest, RecordCueJump) {
+    auto mgr = MacroManager();
+    ChannelHandle handle = ChannelHandleFactory().getOrCreateHandle("");
+    EXPECT_EQ(mgr.m_macroRecordingState.load(), MacroState::Disabled);
+    mgr.notifyCueJump(handle, 0, 1);
+    EXPECT_EQ(mgr.m_activeChannel, nullptr);
+    mgr.m_macroRecordingState.store(MacroState::Armed);
+    mgr.notifyCueJump(handle, 0, 1);
+    EXPECT_EQ(mgr.m_activeChannel->handle(), handle.handle());
+    EXPECT_EQ(mgr.m_recordedMacro.actions[0].position, 0);
+    EXPECT_EQ(mgr.m_recordedMacro.actions[0].target, 1);
```

Comment:
I think there should not be any member variable `m_X` that is accessible from the outside. Make them all private and access them via methods.

Reply:
It isn't. I'm using test friends.

Reply:
Yes, but this will require constant changes every time the macro recorder's internal data format is changed, which is not what we want to test here. Instead, the tests should usually operate on publicly accessible methods.

Reply:
I'd just call notifyCueJump a couple of times and check if the end result is correct, not mid-recording.